tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,327 @@
+        if program_info.find(program_name) != -1:
+            program_status = program_info.split()[1].strip()
+            if program_status == "RUNNING":
+                program_pid = int(program_info.split()[3].strip(' ,'))
+            break
+
+    if program_status:
+        logger.info("Found program {} in the {} state with pid {}".format(program_name, program_status, program_pid))
+            
+    return program_status, program_pid
+
+def get_autorestart_container_and_state(duthost):
+    """
+    @summary: Get container names and its autorestart states by analyzing 
+              the command output of "show container feature autorestart"
+    @return:  A dictionary where keys are the names of containers which have the 
+              autorestart feature implemented and values are the autorestart feature
+              state for that container
+    """
+    autorestart_containers = {}
+
+    show_cmd_output = duthost.shell(CMD_CONTAINER_FEATURE_AUTORESTART)
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip(' \n')
+        if container_state in ["enabled", "disabled"]:
+           autorestart_containers[container_name] = container_state
+
+    return autorestart_containers
+
+def get_disabled_container_list(duthost):
+    """
+    @summary: Get the container/service names which are disabled by default
+    @return: A list includes the names of disabled container/service
+    """
+    disabled_containers = []
+    container_list = []
+
+    redis_cli_keys_output = duthost.shell("redis-cli -n 4 keys 'FEATURE|*'")
+    for line in redis_cli_keys_output["stdout_lines"]:
+        container_list.append(line.strip(' \n').split('|')[1])
+
+    for container_name in container_list:
+        redis_cli_values_output = duthost.shell("redis-cli -n 4 hgetall 'FEATURE|{}'".format(container_name))
+        values = redis_cli_values_output["stdout_lines"]
+        if "status" in values and (values.index("status") + 1 < len(values)) \
+                and values[values.index("status") + 1] == "disabled":
+            disabled_containers.append(container_name)
+
+    return disabled_containers
+
+def is_container_running(duthost, container_name):
+    """
+    @summary: Decide whether the container is running or not
+    @return:  Boolean value. True represents the container is running
+
+    """
+    is_running = duthost.shell("docker inspect -f \{{\{{.State.Running\}}\}} {}".format(container_name))
+    return is_running["stdout_lines"][0].strip(' \n')
+
+def get_program_status(duthost, container_name, program_name):
+    """
+    @summary: Return the status of a program in the specified container
+    @return:  "RUNNING" or "EXITED" represents the program is in the running
+              or exited status
+    """
+    program_status = None
+
+    process_list = duthost.shell("docker exec {} supervisorctl".format(container_name))
```

Comment:
Just running `supervisorctl` will start a supervisorctl shell. I don't think you want to do that. You should run `supervisorctl status` instead.